src/plans/D3MCompoundPlan.sol

```diff
@@ -0,0 +1,187 @@
+
+pragma solidity ^0.8.14;
+
+import "./ID3MPlan.sol";
+
+// https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/CErc20.sol#L14
+interface CErc20Like {
+    function totalBorrows()           external view returns (uint256);
+    function totalReserves()          external view returns (uint256);
+    function interestRateModel()      external view returns (address);
+    function getCash()                external view returns (uint256);
+    function underlying()             external view returns (address);
+}
+
+// https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/BaseJumpRateModelV2.sol#L10
+interface InterestRateModelLike {
+    function baseRatePerBlock()       external view returns (uint256);
+    function kink()                   external view returns (uint256);
+    function multiplierPerBlock()     external view returns (uint256);
+    function jumpMultiplierPerBlock() external view returns (uint256);
+}
+
+contract D3MCompoundPlan is ID3MPlan {
+
+    mapping (address => uint256) public wards;
+    InterestRateModelLike        public tack;
+    uint256                      public barb; // target Interest Rate Per Block [wad] (0)
+
+    CErc20Like public immutable cDai;
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+    event File(bytes32 indexed what, address data);
+
+    constructor(address cDai_) {
+        address rateModel_ = CErc20Like(cDai_).interestRateModel();
+        require(rateModel_ != address(0), "D3MCompoundPlan/invalid-rateModel");
+
+        cDai = CErc20Like(cDai_);
+        tack = InterestRateModelLike(rateModel_);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "D3MCompoundPlan/not-authorized");
+        _;
+    }
+
+    // --- Math ---
+    uint256 internal constant WAD = 10 ** 18;
+    function _wmul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = (x * y) / WAD;
+    }
+    function _wdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = (x * WAD) / y;
+    }
+
+    // --- Admin ---
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "barb") {
+            barb = data;
+        } else revert("D3MCompoundPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+    function file(bytes32 what, address data) external auth {
+        if (what == "tack") tack = InterestRateModelLike(data);
+        else revert("D3MCompoundPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
```

Comment:
Why have a `file` that allows setting an arbitrary `tack` when it needs to be the cDai interest rate model anyway? At a minimum it seems this `file` should `require` the new `tack` be equal to `CErc20Like(cDai_).interestRateModel()`.

Reply:
The new `tack` should be a specific one (i.e a parameter and not fetched from `cDai`) since we only want to support a model that was reviewed by our side.

We can add the `require` you suggested but in any case if there is a mismatch then `active()` check will fail and will cause unwinding (and not supporting the new unknown model).
A secondary reasoning here is that we do usually try to make the `file` functions as flat/minimal as possible.

Let me know if you still think we need the `require` (which will fail the spell but not cause unwinding).

Reply:
Hmmmm I guess not having a `require` gives us some flexibility in case we've reviewed a plan before Compound has officially switched to it and want to go ahead and `file` it. Arguably we should have a whitelist of plans so that we can smoothly migrate with no disruptions, similar to what is done for upgradeable ERC20 gems. So...I argree no `require` needed, but now you need to talk me out of having a whitelist instead of a unique value.

Reply:
Yeah this was also suggested and turned down since we didn't want to support protocols to return to an old supported version without us unwinding. The downside of unwinding and waiting for governance was deemed as not too bad. Personally I think we might want to support the whitelist. If we want to re-open the discussion it should be done in the `v2` PR.